pkg/resource/deploy/snapshot.go

```diff
@@ -127,12 +127,15 @@ func (snap *Snapshot) VerifyIntegrity() error {
 				}
 			}
 
-			if _, has := urns[urn]; has && !state.Delete {
-				// The only time we should have duplicate URNs is when all but one of them are marked for deletion.
-				return errors.Errorf("duplicate resource %s (not marked for deletion)", urn)
+			if _, has := urns[urn]; has && !state.Delete && state.Status == "" {
+				// The only time we should have duplicate URNs is when all but one of them are marked for deletion
+				// or have an operation pending on them.
+				return errors.Errorf("duplicate resource %s (not marked for deletion or pending operation)", urn)
```

Comment:
This makes me a bit nervous: I don't love that we're adding another opportunity for duplicate URNs, even if it is pretty restricted.

As much as I hate to suggest it, what about storing in-flight operations in a separate part of the checkpoint? Technically we're still able to change the V2 snapshot without breaks...

Reply:
FWIW, I had imagined that rather than add duplicate resources, we would set the status on the existing resource (if any).